earthfile2llb/parser/EarthLexer.g4

```diff
@@ -81,7 +81,9 @@ WS_R: WS -> type(WS);
 mode COMMAND_ARGS;
 
 Atom: (RegularAtomPart | QuotedAtomPart)+;
-fragment QuotedAtomPart: ('"' (~'"' | '\\"')* '"');
+fragment QuotedAtomPart: SingleQuotedAtomPart | DoubleQuotedAtomPart;
+fragment SingleQuotedAtomPart: ('\'' (~'\'' | '\\\'')* '\'');
+fragment DoubleQuotedAtomPart: ('"' (~'"' | '\\"')* '"');
```

Comment:
This part seems to make sense, however, you probably also need to eliminate `'` from the `RegularAtomPart` fragment. So something like

```
fragment RegularAtomPart: ~([ \t\r\n\\"\\']) | EscapedAtomPart;
```

(Note that there's also `fragment RegularAtomPart_CAKV` which would need updating).

Reply:
That seems like a good catch! I'll take a look.